File: C++/ex14/string_length_solution.cpp
// String Length Counter (Exercise 14) - Solution
// Find the longest and shortest string from input.

#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    
    vector<string> strings(n);
    for (int i = 0; i < n; i++) {
        cin >> strings[i];
    }
    
    string longest = strings[0], shortest = strings[0];
    
    for (int i = 1; i < n; i++) {
        if (strings[i].length() > longest.length()) {
            longest = strings[i];
        }
        if (strings[i].length() < shortest.length()) {
            shortest = strings[i];
        }
    }
    
    cout << longest.length() << " " << longest << endl;
    cout << shortest.length() << " " << shortest << endl;
    
    return 0;
}

File: C#/ex16/string_length_solution.cs
// String Length Counter (Exercise 16) - Solution
// Find the longest and shortest string from input.

using System;
using System.Collections.Generic;
using System.Linq;

class Program {
    static void Main() {
        int n = int.Parse(Console.ReadLine()!);
        
        List<string> strings = new List<string>();
        for (int i = 0; i < n; i++) {
            string? s = Console.ReadLine();
            if (s != null) {
                strings.Add(s);
            }
        }
        
        if (strings.Count == 0) return;
        
        string longest = strings.OrderByDescending(s => s.Length).First();
        string shortest = strings.OrderBy(s => s.Length).First();
        
        Console.WriteLine($"{longest.Length} {longest}");
        Console.WriteLine($"{shortest.Length} {shortest}");
    }
}
